Mlem/App/Views/Shared/ImageView/LargeImageView.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  LargeImageView.swift
+//  Mlem
+//
+//  Created by Sjmarf on 23/07/2024.
+//
+
+import SwiftUI
+
+// ImageView for use in LargePost. Allows blurring NSFW.
+// In future this view would also handle alt-text display,
+// and anything else that can't go in thumbnail views etc.
+
+struct LargeImageView: View {
+    @Environment(NavigationLayer.self) private var navigation
+    @AppStorage("safety.blurNsfw") var blurNsfw: Bool = true
+
+    let url: URL?
+    var shouldBlur: Bool
+    @State var blurred: Bool = false
+    
+    init(url: URL?, blurred: Bool) {
+        @AppStorage("safety.blurNsfw") var shouldBlur = true
+        self.url = url
+        self.shouldBlur = shouldBlur ? blurred : false
+        self._blurred = .init(wrappedValue: shouldBlur ? blurred : false)
+    }
+    
+    @State private var loading: ImageView.LoadingState = .waiting
+
+    var body: some View {
+        ImageView(url: url, onLoadingStateChange: { loading = $0 })
+            .blur(radius: blurred ? 50 : 0, opaque: true)
+            .clipShape(.rect(cornerRadius: AppConstants.largeItemCornerRadius))
+            .overlay {
+                if blurred {
+                    VStack(spacing: AppConstants.standardSpacing) {
+                        Image(systemName: Icons.warning)
+                            .font(.largeTitle)
+                        Text("NSFW")
+                            .fontWeight(.semibold)
```

Comment:
In v1 this is `.black`--that also aligns with the capsule. IMO we should stick with that design, though maybe that's just because I'm used to it?

Reply:
Sure 👍